roaring/roaring.go

```diff
@@ -395,32 +399,299 @@ func (b *Bitmap) Intersect(other *Bitmap) *Bitmap {
+//
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 1 |___X____________X__________|     |      |___X____________X__________|     |      |___X____________X__________|
+//              ^                             |          _                             |
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 2 |_______X________X______X___|     |      |_______X_______________X___|     |      |_______X_______________X___|
+//                  ^                         |              ^                         |
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 3 |_______X___________________|     |      |_______X___________________|     |      |_______X___________________|
+//                  ^                         |              ^                         |
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 4 |___X_______________________|     |      |___X_______________________|     |      |___X_______________________|
+//              ^                             |          _                             |
+// ------------------------------------------------------------------------------------------------------------------------
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 1 |___X____________X__________|     |      |___X____________X__________|     |      |___X____________X__________|
+//              _                             |                       ^                |                       _
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 2 |_______X_______________X___|     |      |_______X_______________X___|     |      |_______X_______________X___|
+//                  _                         |                              ^         |                              ^
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 3 |_______X___________________|     |      |_______X___________________|     |      |_______X___________________|
+//                  _                         |                                        |
+//          ----------------------------      |      ----------------------------      |      ----------------------------
+// Bitmap 4 |___X_______________________|     |      |___X_______________________|     |      |___X_______________________|
+//              _
+func (b *Bitmap) unionIntoTarget(target *Bitmap, others ...*Bitmap) {
```

Comment:
Thinking about this:

I wonder if it wouldn't make more sense to have this be

`func (target *Bitmap) unionIntoTarget(others ...*Bitmap)`

There's a lot of special cases in the code that come from the first parameter being the special parameter, rather than the receiver.

Reply:
That worked, thanks for the idea